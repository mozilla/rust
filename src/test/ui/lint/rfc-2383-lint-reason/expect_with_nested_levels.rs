// check-pass

#![feature(lint_reasons)]

#[expect(unused_mut, reason = "This should trigger because `unused_mut` was allow")]
//~^ WARNING this lint expectation is unfulfilled [unfulfilled_lint_expectation]
//~| NOTE `#[warn(unfulfilled_lint_expectation)]` on by default
//~| NOTE This should trigger because `unused_mut` was allow
mod foo {
  fn bar() {
    #[allow(unused_mut, reason = "v is unused")]
    let mut v = 0;
  }
}

#[expect(unused_mut, reason = "This should trigger because `unused_mut` is no longer expected")]
//~^ WARNING this lint expectation is unfulfilled [unfulfilled_lint_expectation]
//~| NOTE This should trigger because `unused_mut` is no longer expected
mod oof {
    #[warn(unused_mut, reason = "We no longer expect it in this scope")]
    //~^ NOTE the lint level is defined here
    fn bar() {
        let mut v = 0;
        //~^ WARNING variable does not need to be mutable [unused_mut]
        //~| NOTE We no longer expect it in this scope
    }
}

fn main() {}
