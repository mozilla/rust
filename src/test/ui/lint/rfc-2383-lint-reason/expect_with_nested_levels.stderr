warning: variable does not need to be mutable
  --> $DIR/expect_with_nested_levels.rs:23:13
   |
LL |         let mut v = 0;
   |             ----^
   |             |
   |             help: remove this `mut`
   |
   = note: We no longer expect it in this scope
note: the lint level is defined here
  --> $DIR/expect_with_nested_levels.rs:20:12
   |
LL |     #[warn(unused_mut, reason = "We no longer expect it in this scope")]
   |            ^^^^^^^^^^

warning: this lint expectation is unfulfilled
  --> $DIR/expect_with_nested_levels.rs:5:1
   |
LL | #[expect(unused_mut, reason = "This should trigger because `unused_mut` was allow")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unfulfilled_lint_expectation)]` on by default
   = note: This should trigger because `unused_mut` was allow

warning: this lint expectation is unfulfilled
  --> $DIR/expect_with_nested_levels.rs:16:1
   |
LL | #[expect(unused_mut, reason = "This should trigger because `unused_mut` is no longer expected")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: This should trigger because `unused_mut` is no longer expected

warning: 3 warnings emitted

